front/src/modules/simulationResult/helpers/utils.ts

```diff
@@ -0,0 +1,113 @@
+          ...signal,
+          position_start: cutPositionStart,
+          time_start:
+            signal.time_start + (signal.time_end - signal.time_start) * interpolationFactor,
+        };
+      }
+      if (signal.position_start < cutPositionEnd && signal.position_end > cutPositionEnd) {
+        const interpolationFactor =
+          (signal.position_end - cutPositionEnd) / (signal.position_end - signal.position_start);
+        return {
+          ...signal,
+          position_end: cutPositionEnd,
+          time_end: signal.time_end - (signal.time_end - signal.time_start) * interpolationFactor,
+        };
+      }
+      return signal;
+    })
+    .filter(
+      (signal) => signal.position_start >= cutPositionStart && signal.position_end <= cutPositionEnd
+    );
+
+export const filterSpaceTimeCurvesByPosition = (
+  spaceTimeCurves: ProjectPathTrainResult['space_time_curves'],
+  cutPositionStart: number,
+  cutPositionEnd: number
+): ProjectPathTrainResult['space_time_curves'] =>
+  spaceTimeCurves.map((spaceTimeCurve) => {
+    let { positions, times } = { ...spaceTimeCurve };
+
+    // If the position where we want to cut the curve isn't already in the curve positions,
+    // We need to interpolate the corresponding time and it and the position the curve
+
+    if (positions.at(0)! <= cutPositionStart && !positions.includes(cutPositionStart)) {
+      const interpolatedTime = interpolateValue(
+        { positions, speeds: [], times },
+        cutPositionStart,
+        'times'
+      );
+
+      positions = [cutPositionStart, ...positions];
+      times = [interpolatedTime, ...times];
+    }
+
+    if (positions.at(-1)! >= cutPositionEnd && !positions.includes(cutPositionEnd)) {
+      const interpolatedTime = interpolateValue(
+        { positions, speeds: [], times },
+        cutPositionEnd,
+        'times'
+      );
+
+      positions = [...positions, cutPositionEnd];
+      times = [...times, interpolatedTime];
+    }
+
+    const filteredIndex = positions
+      .map((position, index) =>
+        position >= cutPositionStart && position <= cutPositionEnd ? index : -1
+      )
+      .filter((index) => index !== -1);
+
+    return {
+      positions: filteredIndex.map((index) => positions[index]),
+      times: filteredIndex.map((index) => times[index]),
+    };
+  });
+
+export const filterConflictsByPosition = (
+  conflicts: Conflict[],
+  cutPositionStart: number,
+  cutPositionEnd: number
+) =>
+  conflicts
+    .map(({ spaceStart, spaceEnd, timeStart, timeEnd }) => {
+      if (spaceStart < cutPositionStart && spaceEnd > cutPositionStart) {
+        const interpolationFactor = (cutPositionStart - spaceStart) / (spaceEnd - spaceStart);
+        return {
+          spaceStart: cutPositionStart,
+          spaceEnd,
+          timeStart: timeStart + (timeEnd - timeStart) * interpolationFactor,
+          timeEnd,
+        };
+      }
+      if (spaceStart < cutPositionEnd && spaceEnd > cutPositionEnd) {
```

Comment:
What if both of these conditions are true? That is, what if a conflict extends past the first visible OP _and_ past the last visible OP?

(These cut-off functions are no fun - I'd really rather not have them at all…)

Reply:
I forgot to leave a comment in the code for that but I assumed that even if we leave only a very few waypoints visible, it would be unlikely to have only range of the layer left...

@Tguisnet do you think it could happen often ?